src/main/groovy/net/minecraftforge/gradleutils/GenerateActionsWorkflow.groovy

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+package net.minecraftforge.gradleutils
+
+import groovy.transform.CompileStatic
+import net.minecraftforge.gradleutils.gitversion.GitVersionExtension
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.TaskProvider
+import org.yaml.snakeyaml.DumperOptions
+import org.yaml.snakeyaml.Yaml
+
+import javax.inject.Inject
+
+@CompileStatic
+abstract class GenerateActionsWorkflow extends DefaultTask {
+    private static final String NAME = 'generateActionsWorkflow'
+
+    static TaskProvider<GenerateActionsWorkflow> register(Project project) {
+        register(project, NAME)
+    }
+
+    static TaskProvider<GenerateActionsWorkflow> register(Project project, String name) {
+        project.tasks.register(name, GenerateActionsWorkflow)
+    }
+
+    @Inject
+    abstract ProviderFactory getProviders()
+
+    GenerateActionsWorkflow() {
+        this.description = 'Generates the GitHub Actions workflow file for the project, respecting declared subprojects in Git Version.'
+
+        this.outputFile.convention this.project.rootProject.layout.projectDirectory.file("build_${this.project.name}.yaml")
+
+        this.branch.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.info.branch }
+        this.localPath.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.projectPath }
+        this.paths.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.subprojectPaths.collect { "!${it}/**".toString() } }
+        this.tagPrefix.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.tagPrefix }
+        this.gradleJavaVersion.convention 21
+        this.sharedActionsBranch.convention 'v0'
+    }
+
+    @OutputFile
+    abstract RegularFileProperty getOutputFile()
+
+    @Input
+    @Optional
+    abstract Property<String> getBranch()
+
+    @Input
+    @Optional
+    abstract Property<String> getLocalPath()
+
+    @Input
+    @Optional
+    abstract ListProperty<String> getPaths()
+
+    @Input
+    @Optional
+    abstract Property<String> getTagPrefix()
+
+    @Input
+    abstract Property<Integer> getGradleJavaVersion()
+
+    @Input
+    abstract Property<String> getSharedActionsBranch()
+
+    @TaskAction
+    void exec() throws IOException {
+        var push = ['branches': this.branch.getOrElse('master')] as Map<String, ?>
```

Comment:
Should we filter branches? What branch's workflows take priority?
I would assume if a workflow exists on the branch being pushed, it'd be the one to be run.

Reply:
So, the workflow files are read from the default branch of the repository. This branch in the workflow YAML is what branch is going to have the workflow run on. I'll add that clarification.